inc/zoo/swar/SWAR.h

```diff
@@ -386,16 +386,16 @@ greaterEqual(SWAR<NBits, T> left, SWAR<NBits, T> right) noexcept {
     return ~BooleanSWAR<NBits, T>{static_cast<T>(t)};  // ~(x<y) === x >= y
 }
 
-// In the condition where only MSBs will be on, we can fast lookup with 1 multiply the index of the leftmost byte.
+// In the condition where only MSBs will be on, we can fast lookup with 1 multiply the index of the most signficant byte that is on.
 // This appears to be a mapping from the (say) 256 unique values of a 64 bit int where only MSBs of each 8 bits can be on, but I don't fully understand it.
 // Adapted from TAOCP Vol 4A Page 153 Eq 94.
 template<int NBits, typename T>
 constexpr T
-fastFirstLeftOn(SWAR<NBits, T> test) noexcept {
-    const auto width = sizeof(T) * 8;
-    const auto tval = (1ull<<(width-NBits))-1, bval = (1ull<<(NBits-1))-1;
-    const u64 cval = tval / bval; 
-    return (test.value() * cval) >> (width - NBits);
+indexOfMostSignficantLaneSet(SWAR<NBits, T> test) noexcept {
+    const auto TypeWidth = sizeof(T) * 8;
+    const auto TopVal = (T{1}<<(TypeWidth-NBits))-1, BottomVal = (T{1}<<(NBits-1))-1;
+    const u64 MappingConstant = TopVal / BottomVal;
```

Comment:
This explicit use of u64 is incorrect, `TopVal / BottomVal` is executed as a `T` divided by a `T`.  Why do you need to convert that result to u64?